pom.xml

```diff
@@ -10,15 +10,19 @@
         <!-- Dependency versions -->
         <scala.version>2.11</scala.version>
         <!-- The scala library version must be compatible with scala.version -->
-        <scala-library.version>2.11.7</scala-library.version>
+        <scala-library.version>2.11.8</scala-library.version>
+        <scalatest.version>3.0.0-RC1</scalatest.version>
         <spals.dsa.version>1.0.0</spals.dsa.version>
 
         <!-- Plugin versions -->
+        <maven-surefire-plugin.version>2.19</maven-surefire-plugin.version>
         <scala-maven-plugin.version>3.2.2</scala-maven-plugin.version>
+        <scalatest-maven-plugin.version>1.0</scalatest-maven-plugin.version>
     </properties>
 
     <modules>
         <module>graph</module>
+        <module>graph-test</module>
```

Comment:
That's pretty much it. Just complete separation so test code never makes it into production.